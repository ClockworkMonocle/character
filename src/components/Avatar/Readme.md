For when you want to show an image that represents a user or character.

Without an image
```js
<Avatar
  circle
  altText="AP"
/>
```

With an image
```js
<Avatar
  circle
  src="http://lorempixel.com/350/350"
  altText="AP"
/>
```
With icon
```js
require('../Icon');

<Avatar
  circle
  icon={<Icon name="user-secret" />}
/>
```

Large image and avatar
```js
<Avatar
  circle
  src="http://lorempixel.com/128/128"
  altText="AP"
  size={64}
/>
```

Large image and avatar without the circle shape
```js
<Avatar
  src="http://lorempixel.com/128/128"
  altText="AP"
  size={64}
/>
```
