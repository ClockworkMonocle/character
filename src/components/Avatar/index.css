.avatar {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--grey-85);
  overflow: hidden;
  position: relative;
}

.avatar__circle {
  border-radius: 50%;
}

.avatar_img__48,
.avatar__48 {
  height: 48px;
  width: 48px;
}

.avatar_img__64,
.avatar__64 {
  height: 64px;
  width: 64px;
}

.avatar_img {
  position: absolute;
}
