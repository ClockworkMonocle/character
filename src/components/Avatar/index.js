import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Type from 'components/Type';

import './index.css';

export default function Avatar({ circle, src, altText, size, icon }) {
  const containerCss = cn(
    'avatar',
    `avatar__${size}`,
    circle && `avatar__circle`
  );
  const imgCss = cn('avatar_img', `avatar_img__${size}`);

  return (
    <div className={containerCss}>
      {!icon && <Type>{altText}</Type>}
      {icon}
      {src && <img src={src} className={imgCss} alt={altText} />}
    </div>
  );
}

Avatar.propTypes = {
  /** display with rounded edges */
  circle: PropTypes.bool,
  /** the image src to load */
  src: PropTypes.string,
  /** text to appear in place of the image while it's loading */
  altText: PropTypes.string,
  /** when you want to render a placeholder icon instead of text*/
  icon: PropTypes.node,
  size: PropTypes.oneOf([48, 64]),
};

Avatar.defaultProps = {
  circle: false,
  size: 48,
};
