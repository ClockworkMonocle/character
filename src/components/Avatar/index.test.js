import React from 'react';
import { shallow } from 'enzyme';
import Avatar from './index';

test('<Avatar /> renders image and text', () => {
  expect(shallow(<Avatar src="imgsrc" altText="AP" />)).toMatchSnapshot();
});

test('<Avatar /> renders image and icon', () => {
  expect(
    shallow(<Avatar src="imgsrc" altText="AP" icon={<p>some icon</p>} />)
  ).toMatchSnapshot();
});

test('<Avatar /> renders size and circle', () => {
  expect(
    shallow(
      <Avatar
        circle
        size={64}
        src="imgsrc"
        altText="AP"
        icon={<p>some icon</p>}
      />
    )
  ).toMatchSnapshot();
});
