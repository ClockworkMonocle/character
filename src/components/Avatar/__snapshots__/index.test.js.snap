// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders image and icon 1`] = `
<div
  className="avatar avatar__48"
>
  <p>
    some icon
  </p>
  <img
    alt="AP"
    className="avatar_img avatar_img__48"
    src="imgsrc"
  />
</div>
`;

exports[`<Avatar /> renders image and text 1`] = `
<div
  className="avatar avatar__48"
>
  <Type
    color="black"
    element="span"
    variant="default"
  >
    AP
  </Type>
  <img
    alt="AP"
    className="avatar_img avatar_img__48"
    src="imgsrc"
  />
</div>
`;

exports[`<Avatar /> renders size and circle 1`] = `
<div
  className="avatar avatar__64 avatar__circle"
>
  <p>
    some icon
  </p>
  <img
    alt="AP"
    className="avatar_img avatar_img__64"
    src="imgsrc"
  />
</div>
`;
